==== Transaction Order Dependence ====
SWC ID: 114
Severity: Medium
Contract: Lotto
Function name: withdrawLeftOver()
PC address: 424
Estimated Gas Usage: 2472 - 37083
The value of the call is dependent on balance or storage write
This can lead to race conditions. An attacker may be able to run a transaction after our transaction which can change the value of the call
--------------------
In file: ./repositories/smartbugs-curated/lotto.sol:27

msg.sender.send(this.balance)

--------------------
Initial State:

Account: [CREATOR], balance: 0x0, nonce:0, storage:{}
Account: [ATTACKER], balance: 0x0, nonce:0, storage:{}

Transaction Sequence:

Caller: [CREATOR], calldata: , decoded_data: , value: 0x0
Caller: [SOMEGUY], function: sendToWinner(), txdata: 0x8f1c0585, value: 0x0
Caller: [ATTACKER], function: withdrawLeftOver(), txdata: 0x1846f51a, value: 0x0

==== Unchecked return value from external call. ====
SWC ID: 104
Severity: Medium
Contract: Lotto
Function name: withdrawLeftOver()
PC address: 424
Estimated Gas Usage: 2472 - 37083
The return value of a message call is not checked.
External calls return a boolean value. If the callee halts with an exception, 'false' is returned and execution continues in the caller. The caller should check whether an exception happened and react accordingly to avoid unexpected behavior. For example it is often desirable to wrap external calls in require() so the transaction is reverted if the call fails.
--------------------
In file: ./repositories/smartbugs-curated/lotto.sol:27

msg.sender.send(this.balance)

--------------------
Initial State:

Account: [CREATOR], balance: 0x0, nonce:0, storage:{}
Account: [ATTACKER], balance: 0x0, nonce:0, storage:{}

Transaction Sequence:

Caller: [CREATOR], calldata: , decoded_data: , value: 0x0
Caller: [ATTACKER], function: sendToWinner(), txdata: 0x8f1c0585, value: 0x0
Caller: [SOMEGUY], function: withdrawLeftOver(), txdata: 0x1846f51a, value: 0x0

==== Transaction Order Dependence ====
SWC ID: 114
Severity: Medium
Contract: Lotto
Function name: sendToWinner()
PC address: 543
Estimated Gas Usage: 9296 - 64567
The value of the call is dependent on balance or storage write
This can lead to race conditions. An attacker may be able to run a transaction after our transaction which can change the value of the call
--------------------
In file: ./repositories/smartbugs-curated/lotto.sol:20

winner.send(winAmount)

--------------------
Initial State:

Account: [CREATOR], balance: 0x0, nonce:0, storage:{}
Account: [ATTACKER], balance: 0x0, nonce:0, storage:{}

Transaction Sequence:

Caller: [CREATOR], calldata: , decoded_data: , value: 0x0
Caller: [ATTACKER], function: sendToWinner(), txdata: 0x8f1c0585, value: 0x0

==== Unchecked return value from external call. ====
SWC ID: 104
Severity: Medium
Contract: Lotto
Function name: sendToWinner()
PC address: 543
Estimated Gas Usage: 9296 - 64567
The return value of a message call is not checked.
External calls return a boolean value. If the callee halts with an exception, 'false' is returned and execution continues in the caller. The caller should check whether an exception happened and react accordingly to avoid unexpected behavior. For example it is often desirable to wrap external calls in require() so the transaction is reverted if the call fails.
--------------------
In file: ./repositories/smartbugs-curated/lotto.sol:20

winner.send(winAmount)

--------------------
Initial State:

Account: [CREATOR], balance: 0x0, nonce:0, storage:{}
Account: [ATTACKER], balance: 0x0, nonce:0, storage:{}

Transaction Sequence:

Caller: [CREATOR], calldata: , decoded_data: , value: 0x0
Caller: [ATTACKER], function: sendToWinner(), txdata: 0x8f1c0585, value: 0x0


